src/search1_conversion/convert_result.py

```diff
@@ -167,26 +167,20 @@ def _fetch_narrative_info(results, meta):
     search_params['query'] = {
         'bool': {'should': matches}
     }
-    # Make the query for narratives on ES
-    search_results = es_client.search(search_params, meta)
+
     # Get all the source document objects for each narrative result
-    narrative_hits = [hit['doc'] for hit in search_results['hits']]
     narr_infos = {}
-    for narr in narrative_hits:
-        _id = narr['access_group']
-        if str(_id) not in ws_infos:
-            continue
+    for ws_info in ws_infos.values():
         [workspace_id, workspace_name, owner, moddate,
          max_objid, user_permission, global_permission,
-         lockstat, ws_metadata] = ws_infos[str(_id)]
-        if owner in user_profile_map:
```

Comment:
It seems like this test was dropped.  I'm not sure why it was needed in the first place.  Is there a chance a user could not have a profile?

Reply:
Okay.  I guess my only concern is to make that if some screwed up entry made its way in it doesn't break searches completely.

Reply:
My main concern would be making sure the search doesn't throw a key error if we somehow had some stale entry in.  For example, could we have some old legacy account that predated making people do a profile.  It probably isn't the case though.

Reply:
I'm quite sure we don't have accounts without profiles, but these are loosely coupled systems (auth, user profile), so it is possible to have a user account without a user profile. But like I said in the comment above, defensive code has been added to handle this case - if a user  profile is absent, it will use the user's `username` rather than the user's `realname` from the profile.